### ISSUE TITLE
FR: [Frontend] - GraphQL API Part Search

### ISSUE THREAD

**Issue by palmajano** [aemsites/vg-roadchoice-com#70]
### Context/Description

The search interface is exposed to the Front end by means of 3 GraphQL endpoints (Make filter, Model filter, and Part search including filters) that perform the actual OpenSearch query and return the results in an understandable format for the front end.
Thus, the front end needs to be updated to use the new GraphQL endpoints developed in #62 #63 #64 

### Acceptance Criteria

- Only Preview mode is used which uses the url: [https://search-api-qa-eds.aws.43636.vnonprod.com/search](https://search-api-qa-eds.aws.43636.vnonprod.com/search)
- The front end uses the GraphQL endpoints when searching by Parts:
  - The Make filter is filled by using endpoint #62 
  - The Model filter is filled when a value in the MAke filter is selected by using endpoint #63 
  - The search results are displayed by using endpoint #64 with the selected model and make values  (if any)
- The search results are paginated (12 items per page as currently).
- The results are displayed.
- The old search solution is still available.
- End to end testing for #62, #63, and #64

###This ticket must be released with #69 

### Out of scope

- Deployment to Production (do not publish changes)
- Category facets and search
- - Replace existing "Categories" menu that displays to left of the results with CTA "Looking for a specific category?"  - decision was made to keep the front end as it is today, save changes for phase 2. 
- Categories should not load automatically. User must select CTA to expand the available categories related to the search results. - decision was made to keep the front end as it is today, save changes for phase 2.

**Comment by ZoeBeykirch** [comment]
Once this ticket is done, we can deploy the part search to production

**Comment by cincplug** [comment]
Hi @AsmitKMishra please test this now that it's merged to develop, sorry that it couldn't be sooner.
https://develop--vg-roadchoice-com--aemsites.hlx.page/
Apart from category filters in the left sidebar, we are now getting make and model filter dropdown values from OpenSearch too. In some cases, backend also delivers All makes / All models as a value, but the old hardcoded value is still there to prevent layout shift due to empty dropdown before calling Opensearch

**Comment by AsmitKMishra** [comment]
Testing Completed with #69
